### DISCUSSION TITLE
new plugin: unsupported-videos

### DISCUSSION THREAD

**Discussion by rejetto** [rejetto/hfs#841] [Show and tell]
starting with 0.55, you can install  "unsupported-videos" plugin.
Once configured, it will detect videos not supported by the browser (even AVI), and will make a real-time streaming conversion on the server (using ffmpeg), allowing to play in streaming without downloading.
<img width="422" alt="image" src="https://github.com/user-attachments/assets/5f6ff5f6-4034-4dc6-a047-dd00d1b6970d" />

**Comment by HiCyprus** [comment]
Great! Anticipating....

**Reply by timttran** [reply]
The plugin works but certain mkv file plays with sound, and other don't have sound.  Funny things is that when an mkv file is playing, I don't see FFMpeg is running in Task Manager.  This indicates that there's not real-time transcoding.

**Comment by rejetto** [comment]
@timttran if you play an mkv and don't see ffmpeg, it's because that's already supported by the browser.

for the "no sound", it is possible that the video is supported but the audio is not, and i don't know if the plugin is triggered in this case.
I don't have a file to make a test, at the moment.

**Reply by timttran** [reply]
@rejetto thanks for your quick reply....when I try it with my ipad, it's displaying error msg "unsupported video : converting" and "video conversion failed".  On the server side, I do see FFMPeg runs but it exited right off.  I don't why it failed and I don't see any logs of the error msg.  Please assist when you get a chance...thanks

**Reply by SanokKule** [reply]
In the plugin.js of the plugin delete the 2 slashes near the end at the start of this line:
`//proc.stderr.on('data', x => console.log('ffmpeg:', String(x)))`
You'll see the ffmpeg output and can share it here.

**Reply by timttran** [reply]
@SanokKule ....thanks for the tips.  Here's the log image, I don't see of a way to export the log

![image](https://github.com/user-attachments/assets/a119e568-10e6-4fdf-8dca-6590b853283b)

**Reply by SanokKule** [reply]
Can you try using newer ffmpeg version?

**Reply by timttran** [reply]
sure...let me download a newer ffmpeg version.....also, this is from a partial ffmpeg commandline i got from Task Mgr for the video it tried to transcode

-f mp4 -movflags frag_keyframe+empty_moov -vcodec libx264 -pix_fmtyuv420p -acodec aac -strict -2 -pre

**Reply by timttran** [reply]
thanks SanokKule, it's working now
